type-definitions/immutable.js.flow

```diff
@@ -526,8 +526,14 @@ declare module 'immutable' {
     updateIn(keyPath: ESIterable<any>, notSetValue: any, value: any): Map<K,V>;
     updateIn(keyPath: ESIterable<any>, value: any): Map<K,V>;
 
-    mergeIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
-    mergeDeepIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
+    mergeIn(
+      keyPath: ESIterable<any>,
+      ...iterables: (ESIterable<any> | { [key: string]: any })[]
+    ): Map<K,V>;
```

Comment:
This part is correct. However in flow `Map` and `List` are parametrised (generic). So if you merge `ESIterable<any>` with `Map<string, string>` it becomes `Map<string, any>`, `Map<string, Map<string | number>> would become `Map<string, any>`. 

AFAIK there's no way to overload the declaration for different `V` types so that we would have one for `V: Map` and one for `V: List` case.